src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader_TryGet.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Buffers.Text;
+
+namespace System.Text.Json
+{
+    public ref partial struct JsonUtf8Reader
+    {
+        /// <summary>
+        /// Reads the next JSON token value from the source as a <see cref="string"/>.
+        /// Returns true if the UTF-8 encoded token value can be successfully transcoded
+        /// to a UTF-16 string.
+        /// Returns false if the token type is not JsonTokenType.String or
+        /// JsonTokenType.PropertyName.
+        /// </summary>
+        /// <exception cref="ArgumentException">
+        /// Thrown if invalid byte sequence are detected while transcoding.
+        /// </exception>
+        public bool TryGetValueAsString(out string value)
+        {
+            value = default;
+            if (TokenType != JsonTokenType.String && TokenType != JsonTokenType.PropertyName)
+                return false;
+
+            byte[] valueArray = IsValueMultiSegment ? ValueSequence.ToArray() : ValueSpan.ToArray();
+
+            var utf8 = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+            //TODO: Perform additional validation and unescaping if necessary
+            value = utf8.GetString(valueArray);
```

Comment:
Why do we need to allocate the byte[] and a new encoding for this?  Shouldn't we be able to use a staticly cached encoding?  And for !IsValueMultiSegment, we should be able to just use `utf8.GetString(ValueSpan)`.  Seems like the only allocation we should need is the byte[] in the case of IsValueMultiSegment==true.

Reply:
Good catch. I had previously written it as follows which avoided allocations for the case you mentioned but since this library targeted netstandard originally, we didn't have the necessary (safe) encoding APIs.

```C#
ReadOnlySpan<byte> valueArray = IsValueMultiSegment ? ValueSequence.ToArray() : ValueSpan;
```

Will fix.

> Shouldn't we be able to use a staticly cached encoding?

We want to throw on invalid UTF-8 rather than replace the characters silently based on recommendations by @bartonjs, @GrabYourPitchforks

Reply:
> > Shouldn't we be able to use a staticly cached encoding?

> We want to throw on invalid UTF-8 rather than replace the characters silently

```C#
private static readonly Encoding s_utf8WithErrors = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
```

It doesn't need to be allocated anew each time, and also doesn't need to use the one on Encoding.UTF8 :smile:.

Reply:
> It doesn't need to be allocated anew each time

Yes, I realized that's what @stephentoub was saying after the fact. Thank for clarifying.